### PR TITLE
add table definitions for all ChainLink contract events

### PR DESCRIPTION
**Pull request by x5engine**
0x514910771af9ca656af840dff83e8264ecf986ca
0x79febf6b9f76853edbcbc913e6aae8232cfb9de9 (original)
0x8946a183bfafa95becf57c5e08fe5b7654d2807b
0xf5fff180082d6017036b771ba883025c654bc935
0xe1407bfaa6b5965bad1c9f38316a3b655a09d8a6
0x151445852b0cfdf6a4cc81440f2af99176e8ad08
0x05cf62c4ba0ccea3da680f9a8744ac51116d6231
0x25fa978ea1a7dc9bdc33a2959b9053eae57169b5
0x02d5c618dbc591544b19d0bf13543c0728a3c4ec
0xafce0c7b7fe3425adb3871eae5c0ec6d93e01935
0x79febf6b9f76853edbcbc913e6aae8232cfb9de9 (duplicate)

### CODE REVIEW

**Comment by x5engine** [comment]
seems I can't apply to it on gitcoin
https://gitcoin.co/issue/blockchain-etl/ethereum-etl-airflow/65/4110

File: sqls/events/0x02d5c618dbc591544b19d0bf13543c0728a3c4ec/Aggregator_event_AnswerUpdated.json [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+            ],
+            "name": "AnswerUpdated",
+            "type": "event"
+        },
+        "contract_address": "0x02d5c618dbc591544b19d0bf13543c0728a3c4ec",
+        "field_mapping": {},
+        "type": "log"
+    },
+    "table": {
+        "dataset_name": "Aggregator",
```
Comment (askeluv) [inline sqls/events/0x02d5c618dbc591544b19d0bf13543c0728a3c4ec/Aggregator_event_AnswerUpdated.json]: should be `chainlink` for all files

Comment (x5engine) [reply sqls/events/0x02d5c618dbc591544b19d0bf13543c0728a3c4ec/Aggregator_event_AnswerUpdated.json]: For the dataset_name?

**Comment by x5engine** [comment]
I want to do those changes but I'm going out now

File: sqls/events/0x02d5c618dbc591544b19d0bf13543c0728a3c4ec/Aggregator_event_AnswerUpdated.json (hunk 1 above)
Comment (srisankethu) [reply sqls/events/0x02d5c618dbc591544b19d0bf13543c0728a3c4ec/Aggregator_event_AnswerUpdated.json]: > For the dataset_name?
Yup, the "dataset_name" for all the files should be "chainlink"

Comment (x5engine) [reply sqls/events/0x02d5c618dbc591544b19d0bf13543c0728a3c4ec/Aggregator_event_AnswerUpdated.json]: done

**Comment by x5engine** [comment]
sending the changes now

**Comment by x5engine** [comment]
all done @askeluv :)

**Comment by askeluv** [comment]
Thanks @x5engine but looks like you still have the contract addresses as folders - you should remove that folder level. Please have a look at the other folders in `table_definitions` to see what it should look like.

**Comment by x5engine** [comment]
@askeluv sure I can do that but some of those files have the same name that's why I put them in different folders according to their contract, but I can change it so they are in the same folder, (although what should I renamed to?)

Check this screenshot of this current file
image

**Comment by askeluv** [comment]
As mentioned in the issue description:

For step 2 you may have to manually edit the table definition files if there are multiple contracts with the same name. You can add e.g. _v2 so “contract” becomes “contract_v2”.

**Comment by x5engine** [comment]
totally @askeluv on it :)

**Comment by askeluv** [comment]
You've named the files with the version prefix at the end - and not after the name of the contract. @x5engine can you please just read the issue description properly and look at some other PRs for reference? Thanks.

**Comment by x5engine** [comment]
@askeluv, it's not an issue :) changing it in 10 minutes.

**Comment by x5engine** [comment]
@askeluv done :)

**Comment by x5engine** [comment]
What do you think @askeluv ?

**Comment by askeluv** [comment]
I suppose there are that many Aggregator versions then? I have not looked into the specifics on each contract.

@medvedev1088 could you please make the final review and deploy if all looks good? Thanks.

**Comment by x5engine** [comment]
Yes there is like 7 or 8 version of it. From the contracts given top there and check all of them!

You can see inside the code the contract address too which proof it's unique and generated from the tool you provided.

Thank you

Let me know when I can give you my wallet address

Thanks

**Comment by x5engine** [comment]
@askeluv all good? Everything as you wish?

**Comment by medvedev1088** [comment]
Would make sense instead of the version suffix append the pair suffix. If I understand it correctly each aggregator represents a pair: https://eth-usd-aggregator.chain.link/, e.g.:

0x79fEbF6B9F76853EDBcBc913e6aAE8232cFB9De9 is ETH/USD
0xF5fff180082d6017036B771bA883025c654BC935 is BTC/USD
0x25Fa978ea1a7dc9bDc33a2959B9053EaE57169B5 is EUR/USD
etc.

Having individual Oracles would be useful too. Here https://docs.chain.link/docs/binance-chainlink-ethereum-mainnet I've found Binance, CoinMarketCap, CryptoCompare oracles.

**Comment by medvedev1088** [comment]
@x5engine could you please go over the pairs in "Decentralized Price Reference Data for USD Pairs" section here https://eth-usd-aggregator.chain.link/ and rename/add the table definition files including the pair in the name, e.g. 

`Aggregator_ETH_USD_event_AnswerUpdated.json`.

**Comment by askeluv** [comment]
@x5engine after you've made the change that @medvedev1088 requested, can you please try applying to the bounty again on Gitcoin: https://gitcoin.co/issue/blockchain-etl/ethereum-etl-airflow/65/4110 ? Thanks

**Comment by x5engine** [comment]
I can because you accepted the other guy, so please remove him so I can apply and you can accept me thanks @askeluv

**Comment by x5engine** [comment]
I have applied, so please approve me so u can submit the pr link thanks

**Comment by askeluv** [comment]
Did you get approved now @x5engine ? There's something buggy with Gitcoin. I can't actually you see you applying, but I received an email about it and clicked the "Approve" button in it.

**Comment by x5engine** [comment]
@askeluv done and submitted thanks

**Comment by askeluv** [comment]
Looks like you didn't make the changes requested by @medvedev1088 though?

**Comment by x5engine** [comment]
@askeluv I didn't see that comment! Sure I will make those changes! But don't you think that's a little bit of extra work than the requested initial job?

**Comment by askeluv** [comment]
OK if I add a $10 tip to the payment? It's been a lot of back and forth on this PR.

**Comment by askeluv** [comment]
If you make the change, please make sure you update both the file names and the `table_name` field inside.

**Comment by x5engine** [comment]
Okay sure, I'll do those in a few hours but please check it when you can so I can get paid today :)

Thanks

**Comment by x5engine** [comment]
@askeluv @medvedev1088 okay all done as you requested guys thanks

**Pull request merged**